### ISSUE TITLE
All dimensions get loaded on Server-Startup, and stays loaded even without any player inside it!

### ISSUE THREAD

**Issue by Zilacon** [CrucibleMC/Crucible#29]
Chunks are not unloading.
Worlds are not unloading.
Entities are not spawning (hostile/modded/animals and passives seem to spawn)
![Screenshot_562](https://user-images.githubusercontent.com/3002248/80316172-0de08300-87b1-11ea-88c2-4cb71ffe0744.png)

This is using the test build that was sent to me via dropbox.

NOTE: This is a screenshot of a freshly started server after 8 mins of waiting, no players have joined yet, so it has a low entity/chunk load amount.

**Comment by juanmuscaria** [comment]
Everything is working fine on my server and I'm using the same test build.
What plugin you're using to list your worlds?

**Comment by Zilacon** [comment]
TickProfiler

**Comment by Zilacon** [comment]
Heres essentials, so same report from both forge and bukkit.
![Screenshot_563](https://user-images.githubusercontent.com/3002248/80316262-a414a900-87b1-11ea-938b-545343eb2a74.png)

**Comment by Zilacon** [comment]
And forge itself
![Screenshot_564](https://user-images.githubusercontent.com/3002248/80316285-cc040c80-87b1-11ea-8c2b-f889ac9fb552.png)

**Comment by Zilacon** [comment]
Most of the load on our server, which is destroying the TPS. Is from saving all of those unloaded worlds as far as i can see.
https://timings.aikar.co/?id=b79a7fbcbbba46d2891309d0ea740cc1

**Comment by evernife** [comment]
I have created 10 worlds with MultiVerse.
Stopped the server, than started it again.

All of them were loaded, even though config `keepSpawnInMemory: false`

At least /mv unload still works.

![image](https://user-images.githubusercontent.com/11357801/80322996-e9ab9300-87fe-11ea-946e-fdd59de102be.png)

![image](https://user-images.githubusercontent.com/11357801/80322999-ed3f1a00-87fe-11ea-948a-7ddce1a038ef.png)


After /mv unload
![image](https://user-images.githubusercontent.com/11357801/80323002-f203ce00-87fe-11ea-94e8-5b848357f9e1.png)

**Comment by Zilacon** [comment]
We dont use multiverse. We use multiworld.
Adding nTweaks to the server does unload them, but i want to get rid of that mod as its causing other problems.

**Comment by evernife** [comment]
I am trying to understand the exact source of the problem.

I am making a few more tests.

**Comment by Zilacon** [comment]
I also for the life of me cannot find any hostile mobs. Ive tried tweaking every setting in every config file. Even tried removing multiple mods, and removing all of our plugins. Hostile mobs simply dont generate naturally (in dark areas/night time)

**Comment by evernife** [comment]
Lets keep one bug at a time :D
Better open a new issue.

--------------------------------
I have done the test again, the worlds are not being loaded on server startup, it was a config i missed.

But indeed they are not being unloaded as they should be.

This pic is just after server statup without me using any command.
![image](https://user-images.githubusercontent.com/11357801/80323087-90902f00-87ff-11ea-9f6c-e760b977295e.png)

**Comment by evernife** [comment]
@Zilacon Give us yout  **cauldron.yml**

**Comment by Zilacon** [comment]
I did in the timings report i sent you. It shows all of my config settings.
https://timings.aikar.co/?id=b79a7fbcbbba46d2891309d0ea740cc1

**Comment by juanmuscaria** [comment]
It doesn't send your cauldron.yml

**Comment by evernife** [comment]
Only bukkit.yml and spigot.yml are stored on timings.

**Comment by juanmuscaria** [comment]
We think you didn't disabled the keep world loaded inside cauldron.yml

**Comment by Zilacon** [comment]
```
# This is the main configuration file for Cauldron.
# 
# If you need help with the configuration or have any questions related to Cauldron,
# join us at the IRC or drop by our forums and leave a post.
# 
# IRC: #cauldron @ irc.esper.net ( http://webchat.esper.net/?channel=cauldron )
# Forums: http://cauldron.minecraftforge.net/
# 
# Setting: logging.entity-collision-checks Default: false   # Whether to log entity collision/count checks
# Setting: settings.load-chunk-on-forge-tick Default: false   # Forces Chunk Loading during Forge Server Tick events
# Setting: settings.check-entity-max-speeds Default: false   # Removes any entity that exceeds max speed.
# Setting: settings.chunk-gc-grace-period Default: 0   # Grace period of no-ticks before unload
# Setting: plugin-settings.allow-reload Default: false   # Allow plugins to be reloaded. WARNING - breaks with some mods. We *will not* support this!
# Setting: logging.connection Default: false   # Log connections
# Setting: logging.tick-intervals Default: false   # Log when skip interval handlers are ticked
# Setting: logging.user-login Default: false   # Set true to enable debuggin user's login process
# Setting: world-settings.default.flowing-lava-decay Default: false   # Lava behaves like vanilla water when source block is removed
# Setting: protection.spawn-protect Default: true   # Whether to enable Thermos' all-seeing protection in the spawn world
# Setting: protection.no-fallback-alias Default: true   # Don't allow commands of the format plugin:cmd, the plugin: will be removed (recommended to keep at true)
# Setting: world-settings.default.allow-tnt-punishment Default: true   # TNT ability to push other entities (including other TNTs)
# Setting: logging.dump-threads-on-warn Default: false   # Dump the the server thread on deadlock warning (delps to debug the deadlock)
# Setting: logging.entity-speed-removal Default: false   # Whether to log entity removals due to speed
# Setting: settings.entity-max-speed Default: 100   # Square of the max speed of an entity before removing it
# Setting: settings.check-entity-bounding-boxes Default: true   # Removes a living entity that exceeds the max bounding box size.
# Setting: logging.disabled-warnings Default: false   # Disable warning messages to server admins
# Setting: world-settings.default.infinite-water-source Default: true   # Vanilla water source behavior - is infinite
# Setting: logging.world-leak-debug Default: false   # Log worlds that appear to be leaking (buggy)
# Setting: logging.collision-warn-size Default: 200   # Number of colliding entities in one spot before logging a warning. Set to 0 to disable
# Setting: logging.entity-count-warn-size Default: 0   # Number of entities in one dimension logging a warning. Set to 0 to disable
# Setting: optimized.affinity-locking Default: false   # Whether to enable affinity locking. Very technical usage, recommended for dedicated hosts only. Ask on Discord or GitHub for info on how to set this up properly.
# Setting: world-settings.max-players-visible Default: -1   # How many players will visible in the tab list
# Setting: optimized.ram-load-chunks Default: false   # Loads chunks into the system RAM (experimental). WARNING! ENABLING THIS WILL INCREASE RAM USAGE BY OVER 1GB.
# Setting: world-settings.use-real-names Default: false   # Instead of DIM##, use the world name prescribed by the mod! Be careful with this one, could create incompat with existing setups!
# Setting: logging.entity-death Default: false   # Log when an entity is destroyed (dev)
# Setting: logging.entity-despawn Default: false   # Log when living entities are despawned (dev)
# Setting: logging.chunk-unload Default: false   # Log when chunks are unloaded (dev)
# Setting: plugin-settings.default.remap-plugin-file Default: false   # Remap the plugin file (dev)
# Setting: protection.blocked-cmds Default:    # Contains commands you want to block from being used in-game, you must also include command aliases (e.g. blocked-cmds: /op,/deop,/stop,/restart .
# Setting: protection.instant-removal Default:    # Contains Block IDs that you want to NEVER exist in the world i.e. world anchors (just in case) (e.g. instant-removal: 1,93,56,24 
# Setting: fake-players.do-login Default: false   # Raise login events for fake players
# Setting: logging.entity-spawn Default: false   # Log when living entities are spawned (dev)
# Setting: logging.dump-chunks-on-deadlock Default: false   # Dump chunks in the event of a deadlock (helps to debug the deadlock)
# Setting: optimized.redstone-repeater-update-speed Default: -1   # how many milliseconds the server must ignore before trying repeater updates
# Setting: optimized.redstone-redstoneTorch-update-speed Default: -1   # how many milliseconds the server must ignore before trying redstoneTorch updates
# Setting: settings.load-chunk-on-request Default: true   # Forces Chunk Loading on 'Provide' requests (speedup for mods that don't check if a chunk is loaded
# Setting: settings.entity-bounding-box-max-size Default: 1000   # Max size of an entity's bounding box before removing it (either being too large or bugged and 'moving' too fast)
# Setting: settings.dump-materials Default: false   # Dumps all materials with their corresponding id's
# Setting: logging.chunk-load Default: false   # Log when chunks are loaded (dev)
# Setting: logging.detailed-logging Default: false   # Add stack traces to dev logging
# Setting: logging.dump-heap-on-deadlock Default: false   # Dump the heap in the event of a deadlock (helps to debug the deadlock)
# Setting: debug.thread-contention-monitoring Default: false   # Set true to enable Java's thread contention monitoring for thread dumps

logging:
  entity-collision-checks: false
  connection: false
  tick-intervals: false
  user-login: false
  dump-threads-on-warn: false
  entity-speed-removal: false
  disabled-warnings: false
  world-leak-debug: false
  collision-warn-size: 200
  entity-count-warn-size: 0
  entity-death: false
  entity-despawn: false
  chunk-unload: false
  entity-spawn: false
  dump-chunks-on-deadlock: false
  chunk-load: false
  detailed-logging: false
  dump-heap-on-deadlock: false
  load-chunk-on-forge-tick: false
  check-entity-max-speeds: false
  chunk-gc-grace-period: 0
  entity-max-speed: 100
  check-entity-bounding-boxes: false
  load-chunk-on-request: true
  entity-bounding-box-max-size: 1000
  dump-materials: false
settings:
  load-chunk-on-forge-tick: true
  check-entity-max-speeds: false
  chunk-gc-grace-period: 300
  entity-max-speed: 5
  check-entity-bounding-boxes: false
  load-chunk-on-request: false
  entity-bounding-box-max-size: 1000
  dump-materials: false
plugin-settings:
  allow-reload: true
  default:
    remap-plugin-file: true
    custom-class-loader: true
    debug: false
    remap-nms-v1_7_R4: true
    remap-nms-v1_7_R3: false
    remap-nms-v1_7_R1: false
    remap-nms-v1_6_R3: false
    remap-nms-v1_5_R3: false
    remap-nms-pre: 'false'
    remap-obc-v1_7_R4: true
    remap-obc-v1_7_R3: false
    remap-obc-v1_7_R1: false
    remap-obc-v1_6_R3: false
    remap-obc-v1_5_R3: false
    remap-obc-pre: false
    global-inheritance: true
    plugin-inheritance: true
    remap-reflect-field: true
    remap-reflect-class: true
    remap-allow-future: false
world-settings:
  default:
    flowing-lava-decay: true
    allow-tnt-punishment: true
    infinite-water-source: true
    entity-despawn-immediate: false
    worldgen-TSteelworks-TSBaseWorldGenerator: true
    worldgen-Natura-BaseCropWorldgen: true
    worldgen-Automagy-AutomagyWorldGenerator: true
    worldgen-Metallurgy-WorldGenMetals: true
    worldgen-magicalcrops-WorldGenEssenceEnd: true
    worldgen-Mariculture-WorldGenHandler: true
    worldgen-desertcraft-PamDesertPlantGenerator: true
    worldgen-denseores-WorldGenOres: true
    worldgen-NOVA-WorldGenerator: true
    worldgen-TConstruct-SlimeIslandGen: true
    worldgen-Natura-BaseTreeWorldgen: true
    worldgen-ImmersiveEngineering-IEWorldGen: true
    worldgen-ActuallyAdditions-OreGen: true
    worldgen-TConstruct-TBaseWorldGenerator: true
    worldgen-nevermine-WorldGen: true
    worldgen-harvestcraft-PamBeeGenerator: true
    worldgen-Mekanism-GenHandler: true
    worldgen-essentialcraft-WorldGenManager: true
    worldgen-harvestcraft-PamGardenGenerator: true
    worldgen-GrowthcraftBamboo-WorldGeneratorBamboo: true
    worldgen-Techguns-WorldGenOilHoles: true
    worldgen-witchery-WitcheryWorldGenerator: true
    worldgen-chisel-GeneratorChisel: true
    worldgen-magicalcrops-WorldGenEssence: true
    worldgen-ExtraBees-ModuleGeneration: true
    worldgen-CoFHCore-WorldHandler: true
    worldgen-Thaumcraft-ThaumcraftWorldGenerator: true
    worldgen-factorization-DarkIronOreGenerator: true
    worldgen-harvestcraft-WorldGenPamSalt: true
    worldgen-magicalcrops-WorldGenEssenceNether: true
    worldgen-AncientWarfareStructure-WorldTownGenerator: true
    worldgen-Mariculture-MTWorldGen: true
    worldgen-arsmagica2-AM2WorldDecorator: true
    worldgen-legendgear-StarwellGenerator: true
    worldgen-academycraft-PhaseLiquidGenerator: true
    worldgen-DraconicEvolution-DraconicWorldGenerator: true
    worldgen-kloudspacekloudgear-BaseHexGen: true
    worldgen-GrowthcraftBees-WorldGeneratorBees: true
    worldgen-BigReactors-BRWorldGenerator: true
    worldgen-appliedenergistics2-MeteoriteWorldGen: true
    worldgen-IC2-IC2: true
    worldgen-academycraft-ACWorldGen: true
    worldgen-trevelations-GenExubitura: true
    worldgen-AncientWarfareStructure-WorldStructureGenerator: true
    worldgen-MrTJPCoreMod-SimpleGenHandler$: true
    worldgen-Natura-BaseCloudWorldgen: true
    worldgen-ExtraUtilities-WorldGenEnderLillies: true
    worldgen-lucky-LuckyGenerator: true
    worldgen-MCA-SimpleOreGenerator: true
    worldgen-Forestry-WorldGenerator: true
    worldgen-chocolateQuest-WorldGeneratorNew: true
    worldgen-plantmegapack-PMPWorldGenerator: true
    worldgen-legendgear-AzuriteGenerator: true
    worldgen-Mariculture-GenerationHandler: true
    worldgen-PneumaticCraft-WorldGeneratorPneumaticCraft: true
    worldgen-harvestcraft-PamTreeGenerator: true
    worldgen-OreSpawn-OreSpawnWorld: true
    worldgen-InventoryPets-WorldGenInventoryPets: true
    worldgen-voltzengine-OreGenReplace: true
    worldgen-appliedenergistics2-QuartzWorldGen: true
    worldgen-Techguns-OreGenerator: true
    worldgen-GalacticraftCore-OverworldGenerator: true
    worldgen-necromancy-WorldGenerator: true
    worldgen-GalaxySpace-OreGenerator: true
    worldgen-Roguelike-DungeonGenerator: true
    worldgen-qCraft-QuantumOreGenerator: true
  max-players-visible: -1
  use-real-names: false
protection:
  spawn-protect: true
  no-fallback-alias: true
  blocked-cmds: version, plugins, pl, op
  instant-removal: '[]'
optimized:
  affinity-locking: false
  ram-load-chunks: false
  redstone-repeater-update-speed: 5
  redstone-redstoneTorch-update-speed: 5
fake-players:
  do-login: false
debug:
  thread-contention-monitoring: false
config-version: 1
world-environment-settings:
  normal:
    keep-world-loaded: true
  nether:
    keep-world-loaded: true
    enabled: true
  the_end:
    keep-world-loaded: true
    enabled: true
  storage:
    keep-world-loaded: false
  ab:
    keep-world-loaded: false
  hv:
    keep-world-loaded: false
  pc:
    keep-world-loaded: false
  my:
    keep-world-loaded: false
  ir:
    keep-world-loaded: false
  lu:
    keep-world-loaded: false
  gd:
    keep-world-loaded: false
  gk:
    keep-world-loaded: false
  ds:
    keep-world-loaded: false
  lb:
    keep-world-loaded: false
  vp:
    keep-world-loaded: false
  rn:
    keep-world-loaded: false
  bt:
    keep-world-loaded: false
  dp:
    keep-world-loaded: false
  lc:
    keep-world-loaded: false
  ll:
    keep-world-loaded: false
  acv:
    keep-world-loaded: false
  cl:
    keep-world-loaded: false
  cr:
    keep-world-loaded: false
  cd:
    keep-world-loaded: false
  cp:
    keep-world-loaded: false
    enabled: true
  im:
    keep-world-loaded: false
  sh:
    keep-world-loaded: false
  endoftime:
    keep-world-loaded: false
    enabled: true
  underdark:
    keep-world-loaded: false
    enabled: true
  orespawn:
    keep-world-loaded: false
    enabled: true
  orespawn2:
    keep-world-loaded: false
    enabled: true
  orespawn3:
    keep-world-loaded: false
    enabled: true
  orespawn4:
    keep-world-loaded: false
  orespawn5:
    keep-world-loaded: false
    enabled: true
  orespawn6:
    keep-world-loaded: false
  soulnetworkdimension:
    keep-world-loaded: false
  outer:
    keep-world-loaded: false
    enabled: true
  bedrock:
    keep-world-loaded: false
    enabled: true
  aether:
    keep-world-loaded: false
  dungeons:
    keep-world-loaded: false
  twilightforest:
    keep-world-loaded: true
    enabled: true
  orbit:
    keep-world-loaded: false
    enabled: true
  firstworld:
    keep-world-loaded: false
  tceldrich:
    keep-world-loaded: false
  dreamworld:
    keep-world-loaded: false
    enabled: true
  torment:
    keep-world-loaded: false
    enabled: true
  mirror:
    keep-world-loaded: false
    enabled: true
  asteroids:
    keep-world-loaded: false
    enabled: true
  mars:
    keep-world-loaded: false
    enabled: true
  moon:
    keep-world-loaded: true
    enabled: true
  spectre:
    keep-world-loaded: false
  chroma:
    keep-world-loaded: false
  ceres:
    keep-world-loaded: false
    enabled: true
  kuiper:
    keep-world-loaded: false
    enabled: true
  mercury:
    keep-world-loaded: false
    enabled: true
  pluto:
    keep-world-loaded: false
    enabled: true
  venus:
    keep-world-loaded: false
    enabled: true
  callisto:
    keep-world-loaded: false
    enabled: true
  enceladus:
    keep-world-loaded: false
    enabled: true
  europa:
    keep-world-loaded: false
    enabled: true
  ganymede:
    keep-world-loaded: false
    enabled: true
  io:
    keep-world-loaded: false
    enabled: true
  miranda:
    keep-world-loaded: false
    enabled: true
  titan_we:
    keep-world-loaded: false
    enabled: true
  triton:
    keep-world-loaded: false
    enabled: true
  pocketplane:
    keep-world-loaded: false
    enabled: true
  mercuryorbit:
    keep-world-loaded: false
  venusorbit:
    keep-world-loaded: false
  ceresorbit:
    keep-world-loaded: false
  marsorbit:
    keep-world-loaded: false
  jupiterorbit:
    keep-world-loaded: false
    enabled: true
  saturnorbit:
    keep-world-loaded: false
  uranusorbit:
    keep-world-loaded: false
  neptuneorbit:
    keep-world-loaded: false
  plutoorbit:
    keep-world-loaded: false
  erisorbit:
    keep-world-loaded: false
  kepler22borbit:
    keep-world-loaded: false
  eris:
    keep-world-loaded: false
    enabled: true
  jupiter:
    keep-world-loaded: false
    enabled: true
  kepler22b:
    keep-world-loaded: false
    enabled: true
  neptune:
    keep-world-loaded: false
    enabled: true
  saturn:
    keep-world-loaded: false
    enabled: true
  uranus:
    keep-world-loaded: false
    enabled: true
  deimos:
    keep-world-loaded: false
    enabled: true
  iapetus:
    keep-world-loaded: false
    enabled: true
  oberon:
    keep-world-loaded: false
    enabled: true
  phobos:
    keep-world-loaded: false
    enabled: true
  rhea:
    keep-world-loaded: false
    enabled: true
  titan:
    keep-world-loaded: false
    enabled: true
  titania:
    keep-world-loaded: false
    enabled: true
```

**Comment by evernife** [comment]
Most mod worlds use the **ENVIRONMENT NORMAL**
And this, in your case, is set to keep loaded.

Turn it off to test it :D

```
world-environment-settings:
  normal:
    keep-world-loaded: false
```

Note: Your Overworld (DIM 0) is NORMAL as well, maybe use a plugin like Multiverse and its config "keep-spawn-in-memory" will keep it loaded.

**Comment by Zilacon** [comment]
multiverse is too bloated for our taste. Its locked down to a mc version based on its build. Where as multiworld can literally be plopped onto any mc version at its newest version. Multiverse has also caused biomes to not generate, flora and fauna to not generate, dungeons to not spawn, and even crash the server all together in the past.

Last time i attempted to seek help to them about multiverse using up 4+GB of ram in a huge memory leak, they told me to stop using a modded server then, and that would solve my problems.

**Comment by evernife** [comment]
Ok, any WorldManager plugin may have a simple config like that. Doesn't Multiworld have it?

Besides, test if the worlds stop being loaded with that config.

> world-environment-settings:
>   normal:
>     keep-world-loaded: false

**Comment by Zilacon** [comment]
Id have to do that later, ive already done like 12 restarts and ive pissed off my player base lol.

**Comment by Zilacon** [comment]
![Screenshot_565](https://user-images.githubusercontent.com/3002248/80328499-5c634100-87f4-11ea-8aef-ca6c19d70173.png)
Just wanted to show what happens over time as the servers online...
The chunks pile up and refuse to unload, which then piles up entities and tile entities.
(the worlds are unloaded cause nTweaks is unloading them)

No i haven't changed the config yet. Probably end up doing that tomorrow.

**Comment by evernife** [comment]
My suggestion is to disable the grace period.

The way the grace period cauldron works is not the best one if i may say.

But lag is part of minecraft servers....

You need ot find out exacly what is causing your lag, like using programs like Spark.

> In cauldron.yml
>
>   chunk-gc-grace-period: 0

The amount of tiles is signficant to certain point.
Sometimes a single mod (or even a single block) can be causing a lot of lag.

Install spark and run it for like 1h and see the functions that are causing more lag.

The pic bellow is my server, 20 players on. 

![image](https://user-images.githubusercontent.com/11357801/80329389-89c1e600-8818-11ea-825a-c30e702f7e22.png)

**Comment by Zilacon** [comment]
Changed the config, removed ntweaks, worlds still didnt unload.
But changing the gc grace period drastically helped with unloading chunks/te's. Its always a low amount now, which is fantastic.

So chunks unloading is solved, but worlds unloading and mobs spawning are still mysteries.

**Comment by evernife** [comment]
I can not reproduce the worlds not unloading anymore. 
(After i disabled that cauldron.yml config, i can not reproduce the bug.)

I have created a local server, with just Crucible and Multiverse (you can use your world manager plugin as well).

I created 10 worlds, and when i enter and leave the worlds, they do unload.

-----------------------

Here is what i did.

0- 10 Dimensions created with a plugin, 4 of them have "LoadOnStartup" enabled, 6 don't.


1- Logged in Dimension 0 - Overworld (called Mundo in the pic bellow)
2- Teleported to World Test4
3- DIM0 (Mundo) got unloaded after a few seconds.
4- Teleported from Test4 to Test5
5- After a few seconds, Test4 got unloaded.

![image](https://user-images.githubusercontent.com/11357801/80419608-d5739e80-88af-11ea-9c11-fad26b706aa3.png)

**Comment by juanmuscaria** [comment]
I know it has been a while, but just now I had enough free time to rewrite all our commit into the latest thermos source, can you see if the issue persists in the version 4.0?

**Comment by juanmuscaria** [comment]
I'll be considering this as solved since most weird chunk and world related bugs where fixed after 4.0+
If those problems starts occurring again we can reopen the issue.
